### PR TITLE
[ISSUE #2783] Make a judgment on the failure of nacos registration

### PR DESCRIPTION
**Pull request by gaochuang98**
// Describe your PR here; eg. Fixes #issueNo

Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-register-center/shenyu-register-client/shenyu-register-client-nacos/src/main/java/org/apache/shenyu/register/client/nacos/NacosClientRegisterRepository.java [hunk 1]
```diff
@@ -157,10 +157,13 @@ private synchronized void registerConfig(final String rpcType,
         String configName = RegisterPathConstants.buildServiceConfigPath(rpcType, contextPath);
         try {
             final String defaultGroup = NacosPathConstants.GROUP;
-            configService.publishConfig(configName, defaultGroup, GsonUtils.getInstance().toJson(metadataCache));
+            if (configService.publishConfig(configName, defaultGroup, GsonUtils.getInstance().toJson(metadataCache))) {
+                LOGGER.info("register metadata success: {}", metadata.getRuleName());
```
Comment (KevinClair) [inline shenyu-register-center/shenyu-register-client/shenyu-register-client-nacos/src/main/java/org/apache/shenyu/register/client/nacos/NacosClientRegisterRepository.java:161]: When will it be false?

Comment (gaochuang98) [reply shenyu-register-center/shenyu-register-client/shenyu-register-client-nacos/src/main/java/org/apache/shenyu/register/client/nacos/NacosClientRegisterRepository.java:161]: nacos forbids registration
so when registration fails 
 like iuuse#2710  
registration fails , but out sucess

Comment (KevinClair) [reply shenyu-register-center/shenyu-register-client/shenyu-register-client-nacos/src/main/java/org/apache/shenyu/register/client/nacos/NacosClientRegisterRepository.java:161]: Maybe nacos enable [authentication](https://nacos.io/zh-cn/docs/auth.html), add username and password in the configuration, then try it again.

Comment (gaochuang98) [reply shenyu-register-center/shenyu-register-client/shenyu-register-client-nacos/src/main/java/org/apache/shenyu/register/client/nacos/NacosClientRegisterRepository.java:161]: However, if the registration fails, the registration success log is still output.I think this is also a problem

Comment (KevinClair) [reply shenyu-register-center/shenyu-register-client/shenyu-register-client-nacos/src/main/java/org/apache/shenyu/register/client/nacos/NacosClientRegisterRepository.java:161]: Throw exception when register fail.

Comment (gaochuang98) [reply shenyu-register-center/shenyu-register-client/shenyu-register-client-nacos/src/main/java/org/apache/shenyu/register/client/nacos/NacosClientRegisterRepository.java:161]: will not throw exception，is only false.
look iuuse#2710  log。
output  `register metadata success `  。But actually the registration failed

Comment (KevinClair) [reply shenyu-register-center/shenyu-register-client/shenyu-register-client-nacos/src/main/java/org/apache/shenyu/register/client/nacos/NacosClientRegisterRepository.java:161]: I mean you should throw exception here.

**Comment by KevinClair** [comment]
Makesure your local test passed `mvn clean install -Dmaven.javadoc.skip=true` before push code to origin branch, no need to close and reopen pr.

**Review by KevinClair — APPROVED** [review]

**Pull request merged**